src/components/FactSearchLists/FactRecentSearchLists.tsx

```diff
@@ -0,0 +1,48 @@
+import { Box, Typography, List } from "@mui/material";
+import React from "react";
+import { FactSearchListsPropsTypes } from "@forkfacts/models";
+import { ViewMoreListBtn, FactlListItem } from "@forkfacts/components";
+import { useStyles } from "./styles";
+
+const FactSearchLists: React.FC<FactSearchListsPropsTypes> = ({
+  recentLists,
+  grouped,
+  groupLists,
+  onSelectItem,
+  onViewMore,
+}) => {
+  const styles = useStyles();
+
+  return (
+    <Box className={styles.root}>
+      {grouped ? (
+        <Box>
+          {groupLists!.map((item, index) => (
+            <List key={index} className={styles.groupWrapper}>
+              <Typography color="text.secondary" component="div" className={styles.groupTitle}>
+                {item.groupTitle}
+              </Typography>
+              <Box>
+                {item.listItems.map((item, index) => (
+                  <FactlListItem key={index} item={item} onSelectItem={onSelectItem} />
+                ))}
+              </Box>
+              <ViewMoreListBtn onViewMore={onViewMore!} />
```

Comment:
Absolutely remove `!`. 
! operator tells the compiler to ignore the possibility of it being undefined.
You must understand why and when you are using `i` not just to ignore TypeScript compilation errors.

Read the TypeScript documentation to understand why not-null operator is used. You cannot use it in this context.

Reply:
You have not written your `ViewMoreListBtn` correctly and that is why you need to hack it around it and use `!` operator. See comments on the  `ViewMoreListBtn`